Add HubMetdata

davidfowl opened the pull request:
- Added HubMetadata to detect the hub type on Endpoints
- Added a test

Fixes https://github.com/aspnet/AspNetCore/issues/7783

davidfowl committed:
Add HubMetdata
- Added HubMetadata to detect the hub type on Endpoints
- Added a test

davidfowl committed:
Updated the ref assembly

BrennanConroy commented on src/SignalR/server/Core/src/HubMetadata.cs:
```diff
@@ -0,0 +1,11 @@
+using System;
```
copyright

BrennanConroy commented on src/SignalR/server/SignalR/test/MapSignalRTests.cs:
```diff
@@ -185,6 +185,28 @@ void ConfigureRoutes(IEndpointRouteBuilder routes)
+            {
+                // This "Foo" policy should override the default auth attribute
+                routes.MapHub<AuthHub>("/path")
+                      .RequireAuthorization(new AuthorizeAttribute("Foo"));
```
Assert this?

davidfowl replied:
Let me remove it, it was copy pasta.

rynowak commented on src/SignalR/server/Core/src/HubMetadata.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    public class HubMetadata
+    {
+        public Type HubType { get; set; }
```
Do you really want this to be mutable? Everythign we put in metadata is immutable so far.

davidfowl replied:
Ah good catch.

davidfowl replied:
Bleh, making it immutable means constructor hell though.

rynowak replied:
oh no, a single constructor with a single parameter :trollface:

davidfowl replied:
It's fine for this since HubType is *required* anyways. It's the fact that other potentially optional metadata will need to be get only and will require more ctors

davidfowl committed:
PR feedback
- Added doc comments
- Made HubMetadata immutable
- Updated test to remove auth metadata

davidfowl committed:
Ref assemblies suck

analogrelay reviewed (commented):
Don't have context to review at this time. Looks fine to me though.
